libnetwork/iptables/firewalld.go

```diff
@@ -163,9 +168,15 @@ func checkRunning() bool {
 	if connection == nil {
 		return false
 	}
-	var zone string
-	err := connection.sysObj.Call(dbusInterface+".getDefaultZone", 0).Store(&zone)
-	return err == nil
+	var currentState dbus.Variant
+	currentState, err := connection.sysObj.GetProperty(dbusInterface + ".state")
+
+	var running bool
+	if err == nil {
+		running, _ = regexp.MatchString(".*RUNNING.*", currentState.Value().(string))
```

Comment:
This type-assertion will panic if the concrete type of `currentState.Value()` is not a string, crashing the daemon. The daemon should not crash if firewalld misbehaves.

Reply:
Is there a case where that would happen without an error?

Reply:
It's an RPC call to a foreign process. Anything could happen, including a successful call returning an unexpected result. The dockerd process should not crash _no matter what firewalld does_

Reply:
I added a safer type conversion.